### PR TITLE
gh-100574: add examples/links to the `strptime`/`strftime` docs

### PR DESCRIPTION
**Pull request by ezio-melotti**
* Issue: gh-100574

### CODE REVIEW

File: Doc/library/datetime.rst [hunk 1]
```diff
@@ -737,18 +737,16 @@ Instance methods:
 .. method:: date.strftime(format)
 
    Return a string representing the date, controlled by an explicit format string.
-   Format codes referring to hours, minutes or seconds will see 0 values. For a
-   complete list of formatting directives, see
-   :ref:`strftime-strptime-behavior`.
```
Comment (ezio-melotti) [inline Doc/library/datetime.rst:742]: This `ref` is converted to the title of the section: `strftime() and strptime() Behavior`.  The actual list of format codes is actually a subsection.  Adding a label to the subsection and updating the `ref` might result in a more readable sentence.

File: Doc/library/datetime.rst [hunk 2]
```diff
@@ -2321,6 +2318,15 @@ versus :meth:`strptime`:
 :meth:`strftime` and :meth:`strptime` Format Codes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+These methods accept format codes that can be used to parse and format dates::
+
+   >>> datetime.strptime('31/01/22 23:59:59.999999',
+   ...                   '%d/%m/%y %H:%M:%S.%f')
+   datetime.datetime(2022, 1, 31, 23, 59, 59, 999999)
+   >>>
```
Comment (ezio-melotti) [inline Doc/library/datetime.rst:2326]:
```suggestion
```

Comment (ezio-melotti) [inline Doc/library/datetime.rst:2326]:
```suggestion
```

Comment (ezio-melotti) [inline Doc/library/datetime.rst:2326]:
```suggestion
```

Comment (ezio-melotti) [inline Doc/library/datetime.rst:2326]:
```suggestion
```

File: Doc/library/datetime.rst [hunk 3]
```diff
@@ -2321,6 +2318,15 @@ versus :meth:`strptime`:
...
+   >>> _.strftime('%a %d %b %Y, %I:%M%p')
+   'Mon 31 Jan 2022, 11:59PM'
```
Comment (ezio-melotti) [inline Doc/library/datetime.rst:2328]: In this example I tried to capture as many format codes as possible in a way that makes it easy to find which part corresponds to which code.

File: Doc/library/datetime.rst (hunk 2 above)
Comment (ezio-melotti) [inline Doc/library/datetime.rst:2326]:
```suggestion
```

Comment (ezio-melotti) [inline Doc/library/datetime.rst]:
```suggestion
```

File: Doc/library/datetime.rst [hunk 4]
```diff
@@ -2321,6 +2318,14 @@ versus :meth:`strptime`:
 :meth:`strftime` and :meth:`strptime` Format Codes
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+These methods accept format codes that can be used to parse and format dates::
+
+   >>> datetime.strptime('31/01/22 23:59:59.999999',
+   ...                   '%d/%m/%y %H:%M:%S.%f')
+   datetime.datetime(2022, 1, 31, 23, 59, 59, 999999)
+   >>> _.strftime('%a %d %b %Y, %I:%M%p')
+   'Mon 31 Jan 2022, 11:59PM'
```
Comment (ezio-melotti) [inline Doc/library/datetime.rst:2327]: In this example I tried to capture as many format codes as possible in a way that makes it easy to find which part corresponds to which code.

**Review by hugovk — APPROVED** [review]
Looks good!

https://deploy-preview-100575--python-cpython-preview.netlify.app/library/datetime.html#strftime-and-strptime-format-codes

**Pull request merged**